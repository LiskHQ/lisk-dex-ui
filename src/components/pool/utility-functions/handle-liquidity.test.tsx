import React from 'react';
import HandleLiquidityList from './handle-liquidity-list';
import { fireEvent, render } from "@testing-library/react";


const renderComponent = () => {
    return render(
      <HandleLiquidityList />
    );
  };

  describe("HandleLiquidityList component", () => {
    it("checks if the component matches the snapshot", () => {
        const { container } = renderComponent();
        expect(container).toMatchSnapshot();
    });  

    it("checks if  search bar is working correctly on entring textr", () => {

        const wrapper = renderComponent();
        const downButton = wrapper.getByTestId("arrow-down-button");
        fireEvent.click(downButton);
        const tokenButtonText = wrapper.getByText("Pooled ETH");
        expect(tokenButtonText).toContainHTML("Pooled ETH");        
      });
      it("checks if  search bar is working correctly on entring textr", () => {

        const wrapper = renderComponent();
        const downButton = wrapper.getByTestId("arrow-down-button");
        fireEvent.click(downButton);    
        const increaseLiquidityButton = wrapper.getByTestId("increase-liquidity-button");
        fireEvent.click(increaseLiquidityButton);
        const increaseLiquidityText = wrapper.getByText("Increase Liquidity");
        expect(increaseLiquidityText).toContainHTML("Increase Liquidity");   
      });
});
