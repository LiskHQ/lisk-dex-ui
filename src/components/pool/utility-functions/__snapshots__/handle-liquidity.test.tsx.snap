// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HandleLiquidityList component checks if the component matches the snapshot 1`] = `
<div>
  <div>
    <div
      class="makeStyles-liquidityListPool-3"
    >
      <div>
        <p>
          ETH/LSK
        </p>
      </div>
      <div>
        <i
          data-testid="arrow-down-button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
            />
          </svg>
        </i>
      </div>
    </div>
    <div />
    <div
      class="makeStyles-liquidityList-1"
    >
      <p>
        Do not see a pool you joined
      </p>
      <a>
        Explore existing pools
      </a>
    </div>
  </div>
</div>
`;
